testsuite/tests/typing-immediate/immediate.ml

```diff
@@ -0,0 +1,75 @@
+(* VALID DECLARATIONS *)
```

Comment:
You have no tests where the module signature is separate from the module implementation, for example:


    module type S = sig
      type t [@@immediate]
   end

   module M : S = struct type t = int end
   module M : S = struct type t = string end

   module F (M : S) : S = M

   module M = F (struct type t = int end)
   module M = F (struct type t = string end)